pkg/ebpf/tracee.go

```diff
@@ -825,20 +841,14 @@ func (t *Tracee) initBPF() error {
 	if err != nil {
 		return err
 	}
-
-	err = t.populateBPFMaps()
-	if err != nil {
-		return err
-	}
-
-	if t.config.Capture.NetIfaces != nil || t.config.Debug {
-		for _, iface := range t.config.Capture.NetIfaces {
-			ingressHook, err := t.attachTcProg(iface, bpf.BPFTcIngress, "tc_ingress")
+	if t.config.Capture.NetIfaces != nil || t.config.Filter.InterfaceToTrace != nil {
```

Comment:
this if statememt is no longer needed - you can just loop over the `t.networkInfo.ifacesIndexList` list.